### PR TITLE
couple of fixes (development)

### PR DESCRIPTION
**Pull request by OriBenHur-Any**
- Handle symlink of docker
  installing gravity with base higher or equal to 1.0.19
  the docker binary is left behind and needs to be removed
- fixed the add_local_repo logic
- add kernel pre-check to make sure we can install and output an informative message to the user if 
Fixes we can't

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -844,6 +832,38 @@ function nvidia_drivers_installation() {
...
+    local mod2=${2/-/.}
+    if [[ ${mod} == ${mod2} ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=(${mod}) ver2=(${mod2})
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
```
Comment (OriBenHur-Any) [inline install.sh]: try to fill the first array with zeros if it's shorter the second array

File: install.sh [hunk 2]
```diff
@@ -844,6 +832,38 @@ function nvidia_drivers_installation() {
...
+    local mod=${1//-/.}
+    local mod2=${2/-/.}
+    if [[ ${mod} == ${mod2} ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=(${mod}) ver2=(${mod2})
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
```
Comment (OriBenHur-Any) [inline install.sh]: fill empty fields in ver1 with zeros

File: install.sh [hunk 3]
```diff
@@ -844,6 +832,38 @@ function nvidia_drivers_installation() {
...
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
```
Comment (OriBenHur-Any) [inline install.sh]: `10#i ` is to convert the number to base 10 (log_10) to deal with leading zeros

File: install.sh [hunk 4]
```diff
@@ -844,6 +832,38 @@ function nvidia_drivers_installation() {
...
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
```
Comment (OriBenHur-Any) [inline install.sh]: return 0: means the two versions are equal
return 1: means that the first version the higher the second
return 2: means that the second version is higher than the first

File: clean.sh [hunk 5]
```diff
@@ -223,7 +223,6 @@ function disable_docker {
   fi
 }
 
-
```
Comment (sashas-anyvision) [inline clean.sh]: please remove empty line

File: clean.sh [hunk 6]
```diff
@@ -235,6 +234,12 @@ function purge_docker {
   if [ -d "/var/lib/docker" ]; then
     rm -rf /var/lib/docker
   fi
+  symlinks=$(find /etc/docker -type l)
```
Comment (sashas-anyvision) [inline clean.sh]: please add explanation that this will happen only if gravity is installed in user pc

File: install.sh [hunk 7]
```diff
@@ -503,26 +503,18 @@ function is_tar_files_exists(){
     elif [ -x "$(command -v yum)" ]; then
       if [ "${INSTALL_METHOD}" == "airgap" ] && [ "${NVIDIA_DRIVER_METHOD}" == "host" ]; then
         TAR_FILES_LIST+=("${RHEL_NVIDIA_DRIVER_FILE}")
```
Comment (sashas-anyvision) [inline install.sh]: please rephrase commit comment to "english" :)

File: install.sh [hunk 8]
```diff
@@ -832,6 +832,38 @@ function nvidia_drivers_installation() {
   fi
 }
 
+function vercomp () {
```
Comment (sashas-anyvision) [inline install.sh]: can you please refactor it with something more "readable" 
for example take major, minor, patch for kernel (with whatever method) and compare it to numbers 
take version of gravity and do the same .. 
consider: https://gist.github.com/siddharthkrish/32072e6f97d7743b1a7c47d76d2cb06c

**Pull request merged**